From ea59d612949505d61550cbbe77db644c5467f97a Mon Sep 17 00:00:00 2001
From: Vlad Schiller <vlad-radu.schiller@imgtec.com>
Date: Mon, 4 Sep 2023 15:09:47 +0100
Subject: [PATCH 643/834] pvr: Enable VK_KHR_bind_memory2 extension

The two functions 'vkBindBufferMemory2' and 'vkBindImageMemory2' are aleady
implemented, and it seems that the flag does not need to be enabled for the
tests to pass.

Signed-off-by: Vlad Schiller <vlad-radu.schiller@imgtec.com>
Reviewed-by: Matt Coster <matt.coster@imgtec.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/25779>
---
 docs/features.txt                   | 2 +-
 src/imagination/vulkan/pvr_device.c | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/docs/features.txt b/docs/features.txt
index e607048cd9d..023c44c0e57 100644
--- a/docs/features.txt
+++ b/docs/features.txt
@@ -423,7 +423,7 @@ Vulkan 1.0 -- all DONE: anv, dzn, lvp, nvk, radv, tu, v3dv, vn
 Vulkan 1.1 -- all DONE: anv, lvp, radv, tu, vn
 
   VK_KHR_16bit_storage                                  DONE (anv/gen8+, dzn, hasvk, lvp, radv, tu/a650, v3dv, vn)
-  VK_KHR_bind_memory2                                   DONE (anv, dzn, hasvk, lvp, nvk, radv, tu, v3dv, vn)
+  VK_KHR_bind_memory2                                   DONE (anv, dzn, hasvk, lvp, nvk, pvr, radv, tu, v3dv, vn)
   VK_KHR_dedicated_allocation                           DONE (anv, dzn, hasvk, lvp, nvk, radv, tu, v3dv, vn)
   VK_KHR_descriptor_update_template                     DONE (anv, dzn, hasvk, lvp, nvk, panvk, radv, tu, v3dv, vn)
   VK_KHR_device_group                                   DONE (anv, dzn, hasvk, lvp, nvk, tu, v3dv, vn)
diff --git a/src/imagination/vulkan/pvr_device.c b/src/imagination/vulkan/pvr_device.c
index 7f64610b1d0..303cf346e3d 100644
--- a/src/imagination/vulkan/pvr_device.c
+++ b/src/imagination/vulkan/pvr_device.c
@@ -163,6 +163,7 @@ static void pvr_physical_device_get_supported_extensions(
    struct vk_device_extension_table *extensions)
 {
    *extensions = (struct vk_device_extension_table){
+      .KHR_bind_memory2 = true,
       .KHR_copy_commands2 = true,
       /* TODO: enable this extension when the conformance tests get
        * updated to version 1.3.6.0, the current version does not
-- 
2.42.0

